src/python/pants/engine/isolated_process.py

```diff
@@ -69,6 +70,10 @@ def __init__(
         self.jdk_home = jdk_home
         self.is_nailgunnable = is_nailgunnable
 
+    @memoized_property
+    def env_dict(self) -> Dict[str, str]:
+        return {self.env[i]: self.env[i + 1] for i in range(0, len(self.env), 2)}
```

Comment:
I'm not following what this is used for? Is the `range` for the sake of stable ordering? Dicts have stable ordering already in Python 3.6+.

Also, we probably want this to return `FrozenDict` from `pants.util.frozendict`.

Reply:
`self.env` is a tuple, not a dict. This reconstructs the original input. If we want to use `FrozenDict` we'll have to modify the rust code which handles `ExecuteProcessRequest` and `extern_store_dict` to work with `FrozenDict`.

Reply:
I see. Sounds good to punt on `FrozenDict` if it would require engine changes here.